plugin/awesoMD/plugin.js

```diff
@@ -33,623 +33,678 @@ const yamlRegex = /```(yaml|yml)\n([\s\S]*?)```(\n[\s\S]*)?/g
+                    // ```javascript [25: 1,4-8]   start line numbering at 25,
+                    //                             highlights lines 1 (numbered as 25) and 4-8 (numbered as 28-32)
+                    if (CODE_LINE_NUMBER_REGEX.test(language)) {
+                        let lineNumberOffsetMatch = language.match(CODE_LINE_NUMBER_REGEX)[2]
+                        if (lineNumberOffsetMatch) {
+                            lineNumberOffset = `data-ln-start-from="${lineNumberOffsetMatch.trim()}"`
+                        }
 
-    /**
-     * Given a markdown slide section element, this will
-     * return all arguments that aren't related to markdown
-     * parsing. Used to forward any other user-defined arguments
-     * to the output markdown slide.
-     */
-    function getForwardedAttributes(section) {
-        const attributes = section.attributes
-        const result = []
-
-        for (let i = 0, len = attributes.length; i < len; i++) {
-            const name = attributes[i].name,
-                value = attributes[i].value
-
-            // disregard attributes that are used for markdown loading/parsing
-            if (/data\-(markdown|separator|vertical|notes)/gi.test(name)) continue
-
-            if (value) {
-                result.push(name + '="' + value + '"')
-            } else {
-                result.push(name)
-            }
-        }
+                        lineNumbers = language.match(CODE_LINE_NUMBER_REGEX)[3].trim()
+                        lineNumbers = `data-line-numbers="${lineNumbers}"`
+                        language = language.replace(CODE_LINE_NUMBER_REGEX, '').trim()
+                    }
 
-        return result.join(' ')
-    }
+                    // Escape before this gets injected into the DOM to
+                    // avoid having the HTML parser alter our code before
+                    // highlight.js is able to read it
+                    code = this.escapeForHTML(code)
 
-    /**
-     * Inspects the given options and fills out default
-     * values for what's not defined.
-     */
-    function getSlidifyOptions(options) {
-        const markdownConfig = deck?.getConfig?.().markdown
-
-        options = options || {}
-        options.separator = options.separator || markdownConfig?.separator || DEFAULT_SLIDE_SEPARATOR
-        options.verticalSeparator =
-            options.verticalSeparator || markdownConfig?.verticalSeparator || DEFAULT_VERTICAL_SEPARATOR
-        options.notesSeparator = options.notesSeparator || markdownConfig?.notesSeparator || DEFAULT_NOTES_SEPARATOR
-        options.attributes = options.attributes || ''
-
-        return options
-    }
+                    // return `<pre><code ${lineNumbers} class="${language}">${code}</code></pre>`;
+
+                    return `<pre><code ${lineNumbers} ${lineNumberOffset} class="${language}">${code}</code></pre>`
+                }
+            }
+
+            if (animateLists === true) {
```

Comment:
Is the type of `animatedList` ever checked? If not then I don't think this is good style, because this will be also true for these cases:
- `animateLists='false'`
- `animatedList=2`

Also how does this change relate to the topic of the PR?

Reply:
These are the codes that we copied from the main reveal.js plugin. The `animateLists` variable is true if defined otherwise it is undefined.
This is not actually related to this PR. Maybe the confusion came up because there's a lot of changes in the plugin.js file.
I will revert this change back, since it is the part of code that was copied from reveal.js plugin. :+1: